src/opossum_lib/core/opossum_generation_arguments.py

```diff
@@ -0,0 +1,50 @@
+# SPDX-FileCopyrightText: TNG Technology Consulting GmbH <https://www.tngtech.com>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import logging
+import sys
+from pathlib import Path
+
+from pydantic import BaseModel
+
+from opossum_lib.core.input_file import FileType, InputFile
+
+
+class OpossumGenerationArguments(BaseModel):
+    spdx_files: list[str]
+    scancode_json_files: list[str]
+    opossum_files: list[str]
+    outfile: str
```

Comment:
Could this be a Path already?
Then adding the ending is just a call to `with_suffix` and you don't need to convert later.

Reply:
Actually all of the list[str] could be a list[Path] which is essentially free because click already checked their existence for us.